app/assets/javascripts/discourse/app/lib/transformer.js

```diff
@@ -1,13 +1,67 @@
-import { VALUE_TRANSFORMERS } from "discourse/lib/transformer/registry";
-import { isTesting } from "discourse-common/config/environment";
+import { capitalize } from "@ember/string";
+import {
+  BEHAVIOR_TRANSFORMERS,
+  VALUE_TRANSFORMERS,
+} from "discourse/lib/transformer/registry";
+import { isProduction, isTesting } from "discourse-common/config/environment";
+
+const CORE_TRANSFORMER = "CORE";
+const PLUGIN_TRANSFORMER = "PLUGIN";
+
+export const transformerTypes = Object.freeze({
+  // key and value must match
+  BEHAVIOR: "BEHAVIOR",
+  VALUE: "VALUE",
+});
 
-// add core transformer names
-const validCoreTransformerNames = new Set(
-  VALUE_TRANSFORMERS.map((name) => name.toLowerCase())
-);
-
-// do not add anything directly to this set, use addValueTransformerName instead
-const validPluginTransformerNames = new Set();
+/**
+ * Valid core transformer names initialization.
+ *
+ * Some checks are performed to ensure there are no repeated names between the multiple transformer types.
+ *
+ * The list can be edited in `discourse/lib/transformer/registry`
+ */
+let validTransformerNames;
+
+// Initialize the valid transformer names, notice that this is a self-invoking function that performs some checks
+// to ensure the transformer names are correctly defined, i.e., lowercase and unique.
+// We're not assigning the result directly to `validTransformerNames` because we're using the function
+// `findTransformerInfoByName` which relies on `validTransformerNames` being initialized.
+(() => {
```

Comment:
What's the benefit of having this as an IIFE? Seems like the code could just be in the top level of the module?

Reply:
TBH, don't know what I was thinking.

It can be just code in the top level.

Reply:
Done